PureBasicDebugger/MemoryViewer.pb

```diff
@@ -3,23 +3,128 @@
 ;  Dual licensed under the GPL and Fantaisie Software licenses.
 ;  See LICENSE and LICENSE-FANTAISIE in the project root for license information.
 ;--------------------------------------------------------------------------------------------
+Global Dim MemoryViewer_Chars.s(31)
 
+#MEMORY_VIEW_TABLE_DATA_DEC = 0 
+#MEMORY_VIEW_TABLE_DATA_HEX = 1 
+#MEMORY_VIEW_TABLE_DATA_OCT = 2 
 
-Global Dim MemoryViewer_Chars.s(31)
+Global MemoryViewTableData.i = MemoryIsHex 
 
-; Wrappers to simplify the handling of the different variable types.
-;
+Procedure.s MemoryViewer_GetDataView(ViewTableData) 
+  Select ViewTableData 
+    Case #MEMORY_VIEW_TABLE_DATA_DEC 
+      ProcedureReturn "DEC" 
+    Case #MEMORY_VIEW_TABLE_DATA_HEX
+      ProcedureReturn "HEX"
+    Case #MEMORY_VIEW_TABLE_DATA_OCT 
+      ProcedureReturn "OCT"
+  EndSelect     
+EndProcedure 
+
+Procedure.s OCT(number.q)
+  Protected oct.s=Space(8*SizeOf(Character))
```

Comment:
Are you sure you need to multiply with SizeOf(Character) ? Space() already takes NbChar as parameter

Reply:
#DEBUGGER_GADGET_Memory_ChkformatDataSection needs to be added into DebuggerCommon.pb it got wiped messing around with git which also broke my build. 

no probably don't need the * sizeOf(Character) either in that case, I'll look into it.